src/models/CommonModel.ts

```diff
@@ -81,13 +94,20 @@ export class CommonModel extends CommonSchema<CommonModel>{
             if(mergeTo.type === undefined){
                 mergeTo.type = mergeFrom.type;
             } else {
-                if(!Array.isArray(mergeTo.type)){
-                    mergeTo.type = [mergeTo.type!]; 
+                const addToType = (type : string) => {
+                    if(!mergeTo.type!.includes(type)){
```

Comment:
You should change order of ifs -> first check if types is array and the push, otherwise change type to array, something like:

```ts
if(Array.isArray(mergeTo.type)) {
  mergeTo.type.includes(type) && mergeTo.type.push(type);
} else {
  mergeTo.type = type ? [mergeTo.type, type] : mergeTo.type
}
```

Reply:
But I need to check whether the type are already included in the else branch as well, so I dont think this simplifies anything 😄